### PR TITLE
Support merge_repeated in K.ctc_decode() beam search.

### PR DESCRIPTION
**Pull request by bzamecnik**
### Summary

It allows CTC beam search to disable merging repeated adjacent symbols even if they are separated by a blank symbol and behave consistently with greedy search. `K.ctc_decode()` has a new argument `merge_repeated=False` (instead of hard-coded constant) which is passed to TF. A test is included.

In addition it upgrades deprecated usage of `tf.sparse_to_dense()` in `K.ctc_decode()` (TF was warning about it) and fixes some doc strings (proper case of boolean values) there.

There's a separate test, since we needed different data and to make the whole test as obvious as possible. Possibly we could craft such data that both tests can be merged and parametrized.

### Related Issues

-  #12238 K.ctc_decode() does not pass merge_repeated
  - + older duplicate: #10852 ctc_decode returns inconsistent result when greedy=False
-  #12240 K.ctc_decode() beam search: tf.sparse_to_dense is deprecated

### PR Overview

- [x] This PR requires new unit tests [y/n] (make sure tests are included)
  - YES, added `test_ctc_decode_beam_search_no_merge()`
- [x] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
  - YES, doc of `K.ctc_decode()` is updated
- [x] This PR is backwards compatible [y/n]
  - YES, the default value of `merge_repeated=False` is the same as the previously hard-coded one
- [x] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)
  - YES, it just add one more optional argument `merge_repeated` to `K.ctc_decode()`

### CODE REVIEW

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -4541,14 +4543,11 @@ def ctc_decode(y_pred, input_length, greedy=True, beam_width=100,
         (decoded, log_prob) = ctc.ctc_beam_search_decoder(
             inputs=y_pred,
             sequence_length=input_length, beam_width=beam_width,
-            top_paths=top_paths, merge_repeated=False)
+            top_paths=top_paths, merge_repeated=merge_repeated)
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: This looks like a breaking behavior change. Why does it need to default to True?

Comment (bzamecnik) [reply keras/backend/tensorflow_backend.py]: Oh, seems like a bug, should be `False`. Thanks for the catch! I'll fix that.

Comment (bzamecnik) [reply keras/backend/tensorflow_backend.py]: Fixed. Sorry :)

**Comment by bzamecnik** [comment]
Travis CI job `12058.6` (Python: 3.6, Theano) failed while downloading Miniconda.

**Comment by bzamecnik** [comment]
Is it possible to rerun the CI job or do I need to make some rebase/force push to trigger it?

**Comment by fchollet** [comment]
Restarting tests.

**Comment by bzamecnik** [comment]
Thanks.

**Comment by themurtazanazir** [comment]
@fchollet this feature is not found in the latest? Did we forget to include it at any refactoring time?

**Comment by tuankhai292002** [comment]
@fchollet  merge_repeated not found in the latest! Did you forget to include it

**Pull request merged**